airflow/providers/ssh/hooks/ssh.py

```diff
@@ -283,6 +283,7 @@ def host_proxy(self) -> paramiko.ProxyCommand | None:
         cmd = self.host_proxy_cmd
         return paramiko.ProxyCommand(cmd) if cmd else None
 
+    @cached_property
```

Comment:
I don’t think you can use `cached_property` here since it turns this from a function to a property. `methodtools.lru_cache` would be more appropriate. (Search the code base for some examples.)

Reply:
Yep. Either this or if/else pattern used elsewhere will work - see https://github.com/apache/airflow/pull/40377#discussion_r1650317715